tree/ntuple/v7/test/ntuple_index.cxx

```diff
@@ -0,0 +1,175 @@
+#include "ntuple_test.hxx"
+
+TEST(RNTupleIndex, Basic)
+{
+   FileRaii fileGuard("test_ntuple_index_basic.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fld = model->MakeField<std::uint64_t>("fld");
+
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard.GetPath());
+
+      for (int i = 0; i < 10; ++i) {
+         *fld = i * 2;
+         ntuple->Fill();
+      }
+   }
+
+   auto pageSource = RPageSource::Create("ntuple", fileGuard.GetPath());
+   auto index = ROOT::Experimental::Internal::CreateRNTupleIndex({"fld"}, *pageSource);
+
+   auto ntuple = RNTupleReader::Open("ntuple", fileGuard.GetPath());
+   auto fld = ntuple->GetView<std::uint64_t>("fld");
+
+   for (unsigned i = 0; i < ntuple->GetNEntries(); ++i) {
+      auto fldValue = fld(i);
+      EXPECT_EQ(fldValue, i * 2);
+      EXPECT_EQ(index->GetEntryIndex({&fldValue}), i);
+   }
+}
+
+TEST(RNTupleIndex, SparseSecondary)
+{
+   FileRaii fileGuardMain("test_ntuple_index_sparse_secondary1.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fldEvent = model->MakeField<std::uint64_t>("event");
+
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "primary", fileGuardMain.GetPath());
+
+      for (int i = 0; i < 10; ++i) {
+         *fldEvent = i;
+         ntuple->Fill();
+      }
+   }
+
+   FileRaii fileGuardSecondary("test_ntuple_index_sparse_secondary2.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fldEvent = model->MakeField<std::uint64_t>("event");
+      auto fldX = model->MakeField<float>("x");
+
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "secondary", fileGuardSecondary.GetPath());
+
+      for (int i = 0; i < 5; ++i) {
+         *fldEvent = i * 2;
+         *fldX = static_cast<float>(i) / 3.14;
+         ntuple->Fill();
+      }
+   }
+
+   auto mainNtuple = RNTupleReader::Open("primary", fileGuardMain.GetPath());
+   auto fldEvent = mainNtuple->GetView<std::uint64_t>("event");
+
+   auto secondaryPageSource = RPageSource::Create("secondary", fileGuardSecondary.GetPath());
+   auto index = ROOT::Experimental::Internal::CreateRNTupleIndex({"event"}, *secondaryPageSource);
+   auto secondaryNTuple = RNTupleReader::Open("secondary", fileGuardSecondary.GetPath());
```

Comment:
This pattern of creating a page source, then the index, then using the RNTupleReader seems a bit odd. For now the API is not the main concern since anyway we'll integrate it in the RDF processing. But something to consider in general

Reply:
This is only for testing. Initially, I added a `CreateIndex` method to the `RNTupleReader`, but since we want to use the index only internally (at least for the foreseeable future) I removed it.